src/server/test/db/metasysTests.js

```diff
@@ -0,0 +1,74 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+const { mocha, expect, testDB } = require('../common');
+const path = require('path');
+const Meter = require('../../models/Meter');
+const Reading = require('../../models/Reading');
+const readMetasysData = require('../../services/readMetasysData');
+
+mocha.describe('Metasys readings', () => {
+	mocha.describe('with metasys-duplicate meter', () => {
+		mocha.beforeEach(async () => {
+			conn = testDB.getConnection();
+			await new Meter(undefined, 'metasys-duplicate', null, false, Meter.type.METASYS).insert(conn);
+		//	meter = await Meter.getByName('metasys-duplicate');
+		});
+
+		mocha.it('handles duplicate readings', async () => {
+			conn = testDB.getConnection();
+			const testFilePath = path.join(__dirname, 'data', 'metasys-duplicate.csv');
+			await readMetasysData(testFilePath, 60, 2, false, conn);
+			const count = await Reading.count(conn);
+			expect(count).to.equal(37);
+		});
+
+		mocha.it('handles cumulative readings', async () => {
+			conn = testDB.getConnection();
+			const testFilePath = path.join(__dirname, 'data', 'metasys-duplicate.csv');
+			await readMetasysData(testFilePath, 60, 2, true, conn);
+			const { reading } = await conn.one('SELECT reading FROM readings LIMIT 1');
+			expect(parseInt(reading)).to.equal(280);
+		});
+	});
+	mocha.describe('with metasys-invalid meter', () => {
+		mocha.beforeEach(async () => {
+			conn = testDB.getConnection();
+			await new Meter(undefined, 'metasys-invalid', null, false, Meter.type.METASYS).insert(conn);
+		});
+
+		mocha.it('errors correctly on an invalid file', () => {
+			conn = testDB.getConnection();
+			const testFilePath = path.join(__dirname, 'data', 'metasys-invalid.csv');
+			return expect(readMetasysData(testFilePath, 30, 1, false, conn)).to.eventually.be.rejected;
```

Comment:
Why does this expect `return` instead of `await`?

Reply:
We get better error messages by giving the promise back to Mocha here.